### DISCUSSION TITLE
computational center

### DISCUSSION THREAD

**Discussion by nsrmhjl** [nsrmhjl/ptDtSc#10] [Ideas]
نظر به اهمیت روز افزون مطالعات مبتنی بر محاسبات در علوم زیستی و مسکوت ماندن روند راه اندازی مرکز محاسباتی انستیتو، پیشنهادات خود برای تسهیل و تسریع این امر را به شرح زیر تقدیم می نمایم:
۱- نظارت مستقیم معاونت پژوهشی:
از آنجا که توسعه یک خوشه محاسباتی چه در نوع اولیه آن و چه در نوع پیشرفته دارای اهمیتی فراتر از فعالیت های آموزشی است، پیشنهاد می شود معاونت پژوهشی به صورت مستقیم راهبری این پروژه را به عهده بگیرد. زیرا اولا دارای امکانات اجرایی بیشتری نسبت به مدیریت آموزشی است و ثانیا به دلیل ارتباط مستقیم با مدیران گروه های تحقیقاتی، می تواند دید بهتری نسبت به نیاز های محاسباتی این گروه ها به دست آورد.
۲- استفاده از یک دیتا سنتر داخلی برای نگهداری سرور موجود و همچنین توسعه آن در آینده:
به دلیل محدودیت های فراوان اداری و نبود فضای مناسب برای راه اندازی یک مرکز داده با امکاناتی چون: تامین برق بدون قطعی (UPS)، اینترنت فیبر نوری، تهویه مطبوع با قابلیت دفع گرمای بسیار زیاد تولید شده در اثر کار کردن CPU و GPU با حد اکثر توان و دسترسی فیزیکی کنترل شده، بهتر است در ابتدا سرور و یا سرورهای خریداری شده به یکی از دیتا سنتر های مورد تایید داخلی منتقل شوند و مراحل ابتدایی پیکر بندی سرورها بر اساس نیازهای محاسباتی انستیتو انجام گیرد. در صورت اختصاص فضایی با امکانات مذکور، این سرور های پیکربندی شده و آماده به کار، به داخل انستیتو منتقل می شوند. لازم به ذکر است که حتی در صورت عدم فراهم آمدن این شرایط نیز فعالیت های پژوهشی مرتبط بدون هیچ خللی و از طریق اتصال SSH به سرور قابل انجام خواهد بود.
تجربیات گذشته نشان داده است که پس از استقرار یک دستگاه در یک بخش تحقیقاتی، به مرور زمان کیفیت و کمیت ارائه خدمت توسط این دستگاه تحت تاثیر روابط بین بخشی قرار گرفته و از ضوابط مطلوب فاصله می گیرد. برای پیشگیری از این آسیب، توسعه یک راهکار فرا بخشی و نظام مند برای ارائه خدمت به تمام بخش های تحقیقاتی بسیار حیاتی است. عدم استقرار سرور در یک بخش خاص اولین گام در این مسیر خواهد بود. شایان ذکر است گام های بعدی از طریق پیکر بندی سرور به راحتی قابل دستیابی است و معاونت پژوهشی می تواند قوانین دسترسی، اولویت ها و ... را مستقیما تعیین نماید.
۳-  جذب یک مدیر سیستم:
این مسئله در کارگروه زیست پزشکی سامانه ای مورد تصویب قرار گرفت. اما نه این کارگروه و نه شورای تحصیلات تکمیلی دارای ابزارهای اجرایی لازم جهت پیشبرد این مهم نیستند. این در حالی است که بخش IT انستیتو پاستور دارای پیش زمینه لازم در خصوص مدیریت یک مرکز محاسبات پژوهش محور نمی باشد. به خدمت گرفتن یک مدیر سیستم که همزمان به علوم کامپیوتر و بایوانفورماتیک مسلط باشد، می تواند روند راه اندازی مرکز محاسباتی را عملیاتی نموده و از ابتر شدن آن پیشگیری نماید.
۴- اختصاص بودجه سالیانه برای توسعه مرکز:
اختصاص این بودجه امکان افزایش تعداد سرورها، ظرفیت ذخیره سازی داده و افزودن کارت های گرافیکی برای انجام شبیه سازی دینامیک مولکولی را به مرور برای این مرکز فراهم می نماید که بازدهی آن به مراتب فراتر از خرید رایانه های مجزا از هم و صرف بودجه اجباری برای مانیتور، صفحه کلید، کیس، پاور و ... برای هریک از این رایانه ها خواهد بود که در نهایت صرف کار با سامانه های اداری می شود.

**Comment by AlirezaDostmohammadi** [comment]
سلام دکتر خوبید؟
به نظرم در تایید همون بحث  نظارت مستقیم معاونت پژوهشی قابل ذکره که بیوانفورماتیک و کارهای محاسباتی جذابیت اش به اینه اصلا از فاز آموزش جداست و وارد فاز درمانی، پیش بینی و ساخت شده.
به نظرم هیچ جوره نمیشه گفت که زیر نظر بخش آموزشی باشه. 
بخشی از واکسن هایی که برای کرونا ساخته شد بهتر میدونید به وسیله طراحی محاسباتی دارو بود و اگر رقرار بود به روش های سنتی این امر جلو بره، طبیعتا امروز شاید اصلا واکسنی نداشتیم.
جدای از این ساخت پپتید و مهندسی پروتئیین که بخش پرکاربردی در درمان برخی بیماری ها هست با استفاده از روش های محاسباتی انجام میشه. 
مثلا من خودم زمانی که دانشجو بودم با راهنمایی دکتر عرب و یک تیمی از رویان روی پروتیین PHD کار میکردیم و این پروژه مهندسی پروتئئین اگر فاز اولش به صورت محاسباتی انجام نمیشد، در کشور ما باتوجه به هزینه های بسیار اندک پژوهش اصلا شدنی نبود و بعد اون بود که مهندسی که انجام شده بود تبدیل به کار آزمایشگاهی شد و نتایج خیلی خوبی ازش بدست آورد که جنبه درمانی و کاربردی داره.
یا من زمانی که  تو UCSD تو آزمایشگاه  با دکتر Karen Pierce کار میکردم، تمام تلاش بر این بود که با استفاده از  ML و  NGS بیایم پیش بینی اوتیسم تو کودکان اوتیسمی انجام بدیم. 
و جدای از این نمونه ها موفق کیت های پیش بینی سرطان ساخته شده و نتیجه کارهایی مثل  NGS هست بسیار زیاده.
و  بهتر میدونید scale دیتاهای زیستی و ابزارهای محاسباتی در حدی هست که اصلا بدون سرور جلو نمیره کار. 
اینکه پاستور درباره این موضوع مقاومت میکنه و نمیخواد داشته باشه خیلی عجیبه!

و جدای از این صرفه اقتصادی بالایی داره. چون هزینه های آزمایشگاهی بسیار کاهش میده.  تو همین مورد  PHD ما حداقل  800 جهش مختلف بررسی کردیم در حالی که اگر روش های محاسباتی، ابزاری مثل گرومکس و سرور نبود، هزینه سرسام آور ساخت پپتید ها باعث میشد کار کلا کنسل بشه.

و کشورهای درحال توسعه اینقدر به مسله کارهای محاسباتی اهمیت میدن که نتیجه اش این میشه که کشوری مثل کانادا به عنوان مثال یه شبکه بسیار عظیم تحت عنوان Compute Canada برای کارهای محاسباتی داره.
https://ccdb.computecanada.ca/security/login

**Comment by AlirezaDostmohammadi** [comment]
جسارته دکتر این موارد رو خدمتتون عرض کنم. شما قطعا خیلی بیشتر نسبت به این موضوع آگاه هستید و صرفا برای بحث بود

**Comment by AlirezaDostmohammadi** [comment]
دکتر یه مسله ای  هم هست که خواستم خدمتتون عرض کنم. 
من خیلی اطلاع ندارم روند پاستور به چه صورتیه اما به من چه برای کسب تجربه در حوزه های فنی مرتبط و یا برای سیاحت همیشه علاقه مند به سفر به خارج از کشور هستم و خیلی آزار دهنده است و برام عذاب آوره که بابت هربار خروج از کشور بخوام به مراکز امنیتی و ... پاسخگو باشم و مجوز از پاستور بگیرم و وثیقه بزارم  و ......
یه بار یادمه سر روشن کردن سرور در پاستور در آخر هفته فرمودید که چه پروسه عجیب و غریب و نشدنی داره، به همین خاطر خواستم بپرسم که از این بابت روند پاستور رو اطلاع دارید به چه شکله؟

**Reply by nsrmhjl** [reply]
نه از اون نظر فرقی با بقیه دانشگاه ها نداره. شما کارت پایان خدمتت رو بگیری پاسپورتت مال خودته.

شما UCSD رفتی؟ تو CV که بهم دادی نبود.

**Comment by AlirezaDostmohammadi** [comment]
<div dir="rtl">
دکتر ucsd نرفتم. ۹ ماه به صورت virtual باهاشون کار کردم. برای فعالیت حضوری
مشکلی پیش اومد و نشد در نهایت. بله تو cv که خدمتتون هست، قرار نداده بودم

**Comment by nsrmhjl** [comment]
اوکی.

**Comment by nsrmhjl** [comment]
سلام @AlirezaDoustmohammadi 
من یه ویدیو تو واتساپ برات فرستادم. 
مشخصات دقیقش رو نمی دونم. یادته یه بحثی با هم داشتیم راجع به اجاره جای سرور؟
میشه یه قیمتی برام بگیری؟ ازمون برآورد خواستن.
همینطور میشه یه برآوردی از بودجه مورد نیاز برای توسعه سالیانه یه سرور بدی؟

**Comment by AlirezaDostmohammadi** [comment]
سلام دکتر خوبید؟ ویدیویی برای من نیومده.
در مورد اجازه فضای سرور حتما.
درباره توسعه سالیانه منظورتون هزینه حفط و نگهداری سرورها است یا ارتقا و
اضافه کردن سیستم به مجموعه سرورها است؟

**Comment by AlirezaDostmohammadi** [comment]
سلام دکتر خوبید؟
من در رابطه با سرور مجدد پیگیری کردم، پارس آنلاین کلا بخش تماسش برای امور مربوط به دیتاسنتر از دسترس خارج هست. 
اما حدود هزینه اش به ازای هر یونیت 1 میلیون هست. در حال حاضر برج میلاد هم اجاره فضا میده که اجاره به ازای هر یونیت 1 میلیون و 200 هست

**Reply by nsrmhjl** [reply]
سلام
ممنونم
می دونی آیا مرکز علمی دیگه ای این کار رو کرده. یعنی سرور هاشو ببره دیتا سنتر ؟

**Comment by nsrmhjl** [comment]
سلام @AlirezaDoustmohammadi 
خوبی؟
شما با گلکسی سرور آشنایی داری؟
[galaxyproject](https://training.galaxyproject.org/training-material/)

**Comment by AlirezaDostmohammadi** [comment]
سلام دکتر جان. خوبید؟
متشکرم.
کار نکردم باهاش ولی درباره کاربردهاش شنیدم. دو تا کاربرد اصلی داره گلکسی که یکیش برای  md هست و یکی اش هم برای  ngs. 
توش میشه پایپلاین هم ایجاد کرد. 

شاید بزرگترین مشکل هم این باشه که یه ران محدودیت زمانی داره که طبیعی هم هست داشته باشه.
از طرفی هم گاهی ممکنه تو کار نیاز باشه پایپلاین رو خیلی کاستوم کنیم و یا یه سری ابزار های متفاوت رو اجرا بگیریم و کار کنیم ولی گلکسی اونقدر اجازه اینو نمیده اما بین بچه های  experimental به این خاطر که نیاز نیست خیلی تو ریز بخش  computational قرار بگیرن خیلی محبوبه

**Comment by nsrmhjl** [comment]
میشه خودت روی سرور خودت یک نسخش رو نصب کنی.
حتی میشه با کوبرنتیز نصبش کنی. به این ترتیب محدودیت زمانیش و بقیه چیز هاش کاملا دست ادمین میوفته.
به نظرم برای جایی که می خواد تازه یک سرور محاسباتی راه بندازه و تمام کاربر هاش یول هستن در زمینه برنامه نویسی گزینه مناسبی باشه.

**Comment by AlirezaDostmohammadi** [comment]
حتما دکتر چکش میکنم. روی سرور هایی که من خودم دارم چون همش روش سی پنل نصب
هست نمی تونم نصب کنم. یعنی دسترسی روت ندارم که نصب کنم ولی میتونم روی سرور
خودتون یا سرور مجازی نصب کنم

**Comment by nsrmhjl** [comment]
نه برای خودمون نمی خوام
اما برای پاستور اگر سر عقل بیاد گزینه خوبیه

**Comment by nsrmhjl** [comment]
سلام ای دوست
@AlirezaDoustmohammadi 
برآوردی از قیمت به دست آوردی؟

**Comment by AlirezaDostmohammadi** [comment]
سلام دکتر خوبید؟
راهکار اول رو من کامل نوشتم. لطفا یه چک مینید ببینید خوبه یا نه؟

[راهکار اول.docx](https://github.com/nsrmhjl/ptDtSc/files/11081714/default.docx)
[راهکار اول.pdf](https://github.com/nsrmhjl/ptDtSc/files/11081715/default.pdf)

**Comment by nsrmhjl** [comment]
سلام ای دوست
اینا این حقوقی رو که نوشتی به هیات علمی هایی هم که استخدام می کنن نمیدن.

**Comment by nsrmhjl** [comment]
باید بنویسی حقوق مصوب
عددش رو نمی دونم و خیلی عوامل درش دخیل هست. شاید باید از خود معاون پژوهشی بپرسم.

**Comment by AlirezaDostmohammadi** [comment]
دکتر جان پس من مینویسم حقوق مصوب.

**Comment by AlirezaDostmohammadi** [comment]
[راهکار اول.docx](https://github.com/nsrmhjl/ptDtSc/files/11082146/default.docx)
[راهکار اول.pdf](https://github.com/nsrmhjl/ptDtSc/files/11082148/default.pdf)

**Comment by AlirezaDostmohammadi** [comment]
سلام دکتر خوبید؟
وقتتون بخیر.

تو صحبتی که پیش از عید داشتیم فرمودید 3 تا سناریو نوشته بشه:
1. استفاده از امکانات موجود و استخدام یک فرد برای پشتیبانی نرم افزاری و سخت افزاری
2. ارتقا تنها با کارت گرافیک
3. ارتقا قدرت سرور (هم کارت گرافیک و هم سی پی بو) تا سقف 7 میلیارد تومن
مورد دوم شدنی نیست.
یعنی به سرور فعلی کارت گرافیک هایی که تو بازار ایران هست نمیخوره طبق استعلامی که من گرفتم.
بنابراین به جاش مورد 3 رو در دو کانفیگ مختلف بنویسم.
درسته؟

**Comment by nsrmhjl** [comment]
سلام ای دوست
حال تو چطوره؟
من اونها رو مثال زدم. 
منظورم یک سناریوی حد اقلی، یک سناریوی میانی و یک سناریوی حد اکثری بود.

یکی همون که نوشتی. یکی مثلا با ۳ و نیم میلیارد و یکی با هفت میلیارد.

**Comment by AlirezaDostmohammadi** [comment]
قربانت دکتر جان.
حتما.
به محض باز شدن بازار پیش فاکتور میگیرم و براتون ارسال میکنم.

فقط تو سناریو دوم و سوم حتما کارت گرافیک لحاظ بشه درسته؟

**Comment by nsrmhjl** [comment]
بله و بله

**Comment by AlirezaDostmohammadi** [comment]
حتما دکتر. درخواست پیش فاکتور دادم.

**Comment by AlirezaDostmohammadi** [comment]
سلام دکتر حالتون چطوره خوبید؟
پیش فاکتور بالاخره ارسال شد.

من یه چند مطلب در رابطه با کانفیگ خدمتتون توضیح بدم و بعدش ارسالش کنم.

اول اینکه آخرین ورژن سرور های  HP که این شرکت فرتاک موجود داره سری:
HPE ProLiant DL580 Gen10
هست.

دیتاشیت اش:


[HPE ProLiant DL580 Gen10 Server.pdf](https://github.com/nsrmhjl/ptDtSc/files/11159603/HPE.ProLiant.DL580.Gen10.Server.pdf)


آخرین سری ای که hp زده سری Gen11  هست که شرکت های دیگه تو ایران وارد کردن ولی این شرکت که قبلا با انیسیتو کار کرده، واردش نکرده.

**Comment by AlirezaDostmohammadi** [comment]
من کانفیگ رو بر اساس چیزی که در صفحه 33 نوشته شده که قدرتمند ترین حالت هست، نوشتم با کمی تغییرات که خدمتتون عرض میکنم.

اول در رابطه با cpu :
قوی ترین cpu هایی که برای این سرور قابل استفاده هست:
Intel Xeon-Platinum 8280 (2.7GHz/28-core/205W)
و 
Intel Xeon-Platinum 8276 (2.2GHz/28-core/165W)
عه که هر دو 28 core   و 56 thread دارن.
با توجه به اینکه در اکثر کارها میزان فرکانس از یه مقدار که بالاتر میره دیگه تاثیر زیادی نداره، من 
Intel Xeon-Platinum 8276
انتخاب کردم که به مراتب میزان مصرف برق اش کمتره.

https://ark.intel.com/content/www/us/en/ark/products/192470/intel-xeon-platinum-8276-processor-38-5m-cache-2-20-ghz.html


این سری مال اواسط سال 2019 هست.
میزان کش اش 38.5 مگابایت هست و تا 1 ترابایت رم پشتیبانی میکنه

چیزی که من درخواست دادم، استفاده از 4 cpu در سرور بود که در مجموع تعداد core ها میشه 112 تا و تا 4 ترابایت رم رو پشتیبانی میکنه که اگر در آینده احتیاج به ارتقا رم شد، قطعا دستمون خیلی باز خواهد بود.

**Comment by AlirezaDostmohammadi** [comment]
تو سرور فعلی پاستور ار پردازنده
Intel® Xeon® Processor E7-4850 v4

https://www.intel.com/content/www/us/en/products/sku/93806/intel-xeon-processor-e74850-v4-40m-cache-2-10-ghz/specifications.html
استفاده شده که 16 core بیشتر نداره و با احتساب 4 پردازنده عه روی سرور، کلا تعداد core ها میشه 64 core

تو سری های جدید اینتل از فناوری هایی برای deeplearning هم استفاده میکنه که شاید برای آینده استفاده بشه و این قابلیت در پردازنده  E7-4850 v4 وجود نداره اما در سری 8276 هست

**Comment by AlirezaDostmohammadi** [comment]
در رابطه با کارت گرافیک:
سری های سری Gen10کلا از 4 تا کارت گرافیک پشتیبانی میکنن:

NVIDIA Tesla P40 24GB Module
NVIDIA Tesla V100 PCIe 32GB Module
HPE NVIDIA Quadro RTX6000 GPU
HPE NVIDIA Quadro RTX8000 GPU

و حداکثر از 4 تا gpu میشه استفاده کرد.

چیزی که من به تجربه دیدم اکثرا تو سرور های قوی از V100 استفاده میکنن. به همین خاطر اینو انتخاب کردم.

**Comment by AlirezaDostmohammadi** [comment]
میزان رم رو هم 512 گیگ در نظر گرفتم.
که حالت پایه است و بعدا اگر نیاز بود میشه ارتقا اش داد.
تو سرور فعلی پاستور میزان رم 128 گیگ انگار بود.

**Comment by AlirezaDostmohammadi** [comment]
قیمت پایه:
یعنی 4 تا cpu عهIntel Xeon-Platinum 8276 (2.2GHz/28-core/165W)
+
512 گیگ رم
 + 
یک کارت گرافیک 
NVIDIA Tesla V100 PCIe 32GB Module
شد:

21,350,000,000
ریال
معادل 2 میلیارد و 135 میلیون تومن.

**Comment by AlirezaDostmohammadi** [comment]
و اگر از 4 تا کارت گرافیک استفاده بشه قیمت میشه:
33,800,000,000
ریال معادل 
3 میلیارد و 380 میلیون تومن

**Comment by AlirezaDostmohammadi** [comment]
پیش فاکتوری که برای من ارسال کردن:

[02-FT-02101  انیستیتو پاستور.pdf](https://github.com/nsrmhjl/ptDtSc/files/11159881/02-FT-02101.pdf)

**Comment by AlirezaDostmohammadi** [comment]
دکتر اگه نیازه میتونم از سری Gen11هم پیش فاکتور بگیرم از شرکت های دیگه که قیمت اون مشخص بشه ولی قطعا قیمت بالاتر خواهد بود.

**Comment by AlirezaDostmohammadi** [comment]
تخمینی که دارم اینکه از 4.5 میلیارد به بالادر میاد.
یه نمونه اش تو بازار:
https://saprasanat.net/product/hpe-dl360-gen11-4lff-nc-cto-server-p52498-b21/

**Comment by AlirezaDostmohammadi** [comment]
سری HPE ProLiant DL380 Gen11 ۀخرین سری هست که انگار در ایران موجوده.

[HPE ProLiant DL380 Gen11-a50004307enw.pdf](https://github.com/nsrmhjl/ptDtSc/files/11160023/HPE.ProLiant.DL380.Gen11-a50004307enw.pdf)

ار کارت گرافیک سری A100پشتیبانی میکنه و آخرین نسل پردازنده های Xeon که هر cpu میتونه حداکثر 60 core داشته باشه.

**Comment by nsrmhjl** [comment]
ممنون
پس این سرور با چهار تا کارت گرافیکی و سی پی یو هایی که میگی جدید هستند میشن سه و خورده ای میلیارد.
این میشه سناریوی میانه. سناریوی تاپ چی میشه؟

**Reply by AlirezaDostmohammadi** [reply]
بله دکتر. میشه 3 میلیارد و 380 میلیون تومن. میتونیم 4 تا سناریو بچینیم.
یکی سرور با یه دونه کارت گرافیک که میشه 2 میلیارد و 135 میلیون تومن
یکی با 4 تا کارت گرافیک
یکی هم میتونم از سری gen11 پیش فاکتور بگیرم که اون بالای 5 میلیارد در میاد یا اینکه مثلا بگیم دو تا سرور با 4 تا گرافیک داشته باشیم که میشه حدود 7 میلیارد

**Comment by nsrmhjl** [comment]
جن ۱۱ برای اونایی که دارن مخاسبات کار می کنن چه مزیتی داره؟

**Reply by AlirezaDostmohammadi** [reply]
دکتر اصلی ترین تفاوت در اینکه از چه سری کارت گرافیک و پردازنده هایی پشتیبانی میکنه.
مثلا تو gen10 حداکثر از کارت گرافیک v100پشتیبانی میکنه و حداکثر تعداد core پردازنده هایی که پشتیبانی میکنه 28 تا است.

ولی تو gen 10 plus , gen 11 از سری کارت گرافیک A100 هم پشتیبانی میکنه و پردازند ه هایی که پشتیبانی میکنه هر کدومشون 60 تا core حداکثر میتونه داشته باشه.

**Reply by AlirezaDostmohammadi** [reply]
و حسن اش برای ما میتونه استفاده از کارت گرافیک قوی تر باشه.
البته از اون طرف مصرف برقش خیلی بالاتره

**Comment by AlirezaDostmohammadi** [comment]
البته gen11 الان زنگ زدم، تو بازار موجود نیست.
gen 10 plus 
تو بازار ایران هست که اونم از کارت گرافیک A100  پشتیبانی میکنه و 80 گیگ رم داره.

**Comment by nsrmhjl** [comment]
خیلی خوب پس همین رو دوتاش کن میشه سناریوی تاپ.

**Reply by AlirezaDostmohammadi** [reply]
حتما دکتر.

**Comment by AlirezaDostmohammadi** [comment]
[سناریو دوم.docx](https://github.com/nsrmhjl/ptDtSc/files/11168342/default.docx)
[سناریو دوم.pdf](https://github.com/nsrmhjl/ptDtSc/files/11168343/default.pdf)

**Comment by AlirezaDostmohammadi** [comment]
اوکیه دکتر؟

**Comment by nsrmhjl** [comment]
سلام خیلی خوبه 
فقط آخرش رقم نداره. یعنی فقط موارد هزینه کرد رو نوشتی. یک عدد کلی بنویس. که حاصل جمع سرور با استراتژی قبلی هست.

**Comment by AlirezaDostmohammadi** [comment]
[سناریو دوم.docx](https://github.com/nsrmhjl/ptDtSc/files/11171246/default.docx)
[سناریو دوم.pdf](https://github.com/nsrmhjl/ptDtSc/files/11171247/default.pdf)
پیوست:
[02-FT-02101  انیستیتو پاستور.pdf](https://github.com/nsrmhjl/ptDtSc/files/11171252/02-FT-02101.pdf)

**Comment by AlirezaDostmohammadi** [comment]
[سناریو سوم.docx](https://github.com/nsrmhjl/ptDtSc/files/11177274/default.docx)
[سناریو سوم.pdf](https://github.com/nsrmhjl/ptDtSc/files/11177275/default.pdf)

**Reply by nsrmhjl** [reply]
خیلی ممنون

**Comment by AlirezaDostmohammadi** [comment]
سلامت باشید دکتر جان
